x-pack/plugins/cases/common/types/domain/case/v1.ts

```diff
@@ -51,6 +51,31 @@ export const CaseSettingsRt = rt.strict({
   syncAlerts: rt.boolean,
 });
 
+const customFieldValue = <C extends rt.Mixed>(codec: C) =>
+  rt.strict({ value: rt.union([rt.array(codec), rt.null]) });
+
+const CustomFieldString = rt.strict({
+  key: rt.string,
+  type: rt.literal('string'),
```

Comment:
I think it is more intuitive if we name it `text`. Same for the name `CustomFieldString` -> `CustomFieldText`